core/src/main/scala/org/apache/spark/sql/SnappySession.scala

```diff
@@ -1647,6 +1645,24 @@ class SnappySession(@transient private val sc: SparkContext,
 
   def setPreparedQuery(preparePhase: Boolean, paramSet: Option[ParameterValueSet]): Unit =
     sessionState.setPreparedQuery(preparePhase, paramSet)
+
+  /**
+   * Callback to execute code when a particular property is set.
+   * @param key
+   * @param value
+   */
+  def confCallback(key: String, value: String): Unit = {
+    key match {
+      case "spark.scheduler.pool" =>
+        // set the scheduler pool if the pool exists else throw an exception. 
+        if (sparkContext.getAllPools.exists(_.name == value)) {
+          sparkContext.setLocalProperty(key, value)
+        } else {
+          throw new IllegalArgumentException(s"Invalid Pool $value")
+        }
+      case _ =>
+    }
+  }
```

Comment:
> What about the jobs? For smart connector, they will use what they were using.

Weren't you arguing that users will continue to use SparkContext.setLocalProperty in jobs and smart connector??

> 1. The property is documented as property for JDBC and ODBC clients.

Exactly. And as per your own stance earlier, for other cases users will continue to use SnappyContext.setLocalProperty, no? Or did I misunderstand what you were saying earlier?

> 2. I am not sure if the property is being used for hive servers anymore.

There are two cases here:

a) Either the property "spark.sql.thriftserver.scheduler.pool" is available in Spark JDBC/ODBC and users are using that. In that case we must use the same property. As per the code it is clearly available and not marked deprecated in code but for some reason is not in recent docs.

b) Or that property is not available. In that case users cannot set the pool over JDBC/ODBC in Spark currently. In that case our addition can be quite useful for Spark JDBC/ODBC too and we can use either of the property names. But we must use the SnappyConf way so that then the same property can be used for both Spark Hive JDBC/ODBC (when we add SnappySession support in that) as well as SnappyData JDBC/ODBC.